tests/quick.rs

```diff
@@ -0,0 +1,200 @@
+    S: BuildHasher + Clone,
+{
+    for op in ops {
+        match *op {
+            Add(ref k, ref v) => {
+                a.insert(k.clone(), v.clone());
+                b.entry(k.clone()).or_insert_with(Vec::new).push(v.clone());
+            }
+            Remove(ref k) => {
+                a.empty(k.clone());
+                b.remove(k);
+            }
+        }
+    }
+}
+
+fn assert_maps_equivalent<K, V, S>(a: &ReadHandle<K, V, (), S>, b: &HashMap<K, Vec<V>>) -> bool
+where
+    K: Clone + Hash + Eq + Debug,
+    V: Hash + Eq + Debug + Ord + Copy,
+    S: BuildHasher,
+{
+    assert_eq!(a.len(), b.len());
+    let keys = a.map_into::<_, Vec<K>, _>(|k, _| k.clone());
+    for key in &keys {
+        assert!(b.contains_key(key), "b does not contain {:?}", key);
+    }
+    for key in b.keys() {
+        assert!(a.get(key).is_some(), "a does not contain {:?}", key);
+    }
+    for key in &keys {
+        let mut ev_map_values: Vec<V> = a.get(key).unwrap().iter().copied().collect();
+        ev_map_values.sort();
+        let mut map_values = b[key].clone();
+        map_values.sort();
+        assert_eq!(ev_map_values, map_values);
+    }
+    true
+}
+
+/// quickcheck Arbitrary adaptor -- make a larger vec
+#[derive(Clone, Debug)]
+struct Large<T>(T);
+
+impl<T> Deref for Large<T> {
+    type Target = T;
+    fn deref(&self) -> &T {
+        &self.0
+    }
+}
+
+impl<T> Arbitrary for Large<Vec<T>>
+where
+    T: Arbitrary,
+{
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        let len = g.next_u32() % (g.size() * 10) as u32;
+        Large((0..len).map(|_| T::arbitrary(g)).collect())
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        Box::new((**self).shrink().map(Large))
+    }
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+struct Alphabet(String);
+
+impl Deref for Alphabet {
+    type Target = String;
+    fn deref(&self) -> &String {
+        &self.0
+    }
+}
+
+const ALPHABET: &[u8] = b"abcdefghijklmnopqrstuvwxyz";
+
+impl Arbitrary for Alphabet {
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        let len = g.next_u32() % g.size() as u32;
+        let len = min(len, 16);
+        Alphabet(
+            (0..len)
+                .map(|_| ALPHABET[g.next_u32() as usize % ALPHABET.len()] as char)
+                .collect(),
+        )
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        Box::new((**self).shrink().map(Alphabet))
+    }
+}
+
+#[quickcheck]
+fn operations_i8(ops: Large<Vec<Op<i8, i8>>>) -> bool {
+    let (r, mut w) = evmap::new();
+    let mut reference = HashMap::new();
+    do_ops(&ops, &mut w, &mut reference);
+    w.refresh();
+    assert_maps_equivalent(&r, &reference)
+}
+
+#[quickcheck]
+fn operations_string(ops: Vec<Op<Alphabet, i8>>) -> bool {
+    let (r, mut w) = evmap::new();
+    let mut reference = HashMap::new();
+    do_ops(&ops, &mut w, &mut reference);
+    w.refresh();
+    assert_maps_equivalent(&r, &reference)
+}
```

Comment:
Left out some tests like [keys and values iterators](https://github.com/jonhoo/griddle/blob/master/tests/quick.rs#L179) because evmap doesn't have them. If we want them, I'd be very willing to take a crack at those!

Reply:
No, that seems fine to me!